lib/graphql/schema.rb

```diff
@@ -221,6 +222,22 @@ def types
       end
     end
 
+    # Returns a list of Arguments and Fields referencing a certain type
+    # @param type_name [String]
+    # @return [Hash]
+    def references_to(type_name)
+      rebuild_artifacts unless defined?(@type_reference_map)
+      @type_reference_map.fetch(type_name, [])
+    end
+
+    # Returns a list of Union types in which a type is a member
+    # @param type [GraphQL::ObjectType]
+    # @return [Array<GraphQL::UnionType>] list of union types of which the type is a member
+    def union_memberships(type)
+      rebuild_artifacts unless defined?(@unions)
```

Comment:
Not entirely clear to me if we're expected to be able to call `rebuild_artifacts` more than once (the name kind of suggests yes), but if so it should probably clear `@union_memberships` or that data will be stale/cached?